### PR TITLE
Refactor hamburger menu semantics

### PR DESCRIPTION
**Pull request by peterblazejewicz**
This PR refactors semantics of hamburger menu to use more user friendly/compliant solution (no divs - yeah) - the same as used in:
https://github.com/twbs/bootstrap/blob/master/docs/_includes/nav/main.html#L4
https://github.com/google/web-starter-kit/blob/master/app/index.html#L41

- change markup to use native button
- change style to retain original design and features

The solution was tested - except of IE. The iOS simulator was used to hunt down and fix a bug (line-height needs to be reset for buttons):

menu

One can now trigger menu via any pointing device/software. 

Thanks!

### CODE REVIEW

File: src/stylesheets/_header.scss [hunk 1]
```diff
@@ -108,9 +108,15 @@
 
       @include below(660px) {
         color: #fff;
-        display: block;
+        line-height: inherit;
+        background: transparent;
```
Comment (samccone) [inline src/stylesheets/_header.scss]: in scss we can do

```scss
background: transparent {
  image: none
}
```

Comment (thejameskyle) [inline src/stylesheets/_header.scss]: but why would you want to?

**Comment by peterblazejewicz** [comment]
@samccone @thejameskyle 
refactored and squashed
I have no habit to use nested properties Sass features as I work with Less every-day - so nice tip BTW.
I've changed button markup  to shorter and leaner version I think.

Thanks guys!

File: src/stylesheets/_header.scss (hunk 1 above)
Comment (samccone) [inline src/stylesheets/_header.scss]: It is feature of the language

Comment (thejameskyle) [inline src/stylesheets/_header.scss]: It's gross and confusing though

**Comment by jdaudier** [comment]
@peterblazejewicz I like how you're referencing other sources that have solved the same issue. Keep it up!

**Pull request merged**